docs/data/data-grid/state/state.md

```diff
@@ -66,40 +66,42 @@ Some selectors are yet to be documented.
 
 ## Save and restore the state
 
-> ⚠️ This feature isn't implemented yet. It's coming.
+The current state of the grid can be exported using `apiRef.current.exportState()`.
+It can then be restored by either passing the returned value to the `initialState` prop or to the `apiRef.current.restoreState()` method.
+
+Watch out for controlled models and their callbacks (`onFilterModelChange` if you use `filterModel`, for instance), as the grid will call those callbacks when restoring the state.
+But if the callback is not defined or if calling it does not update the prop value, then the restored value will not be applied.
+
+### Restore the state with `initialState`
+
+You can pass the state returned by `apiRef.current.exportState()` to the `initialState` prop.
+In the demo below, clicking on **Recreate the 2nd grid** will re-mount the 2nd grid with the current state of the 1st grid.
+
+> ⚠️ If you restore the page using `initialState` before the data is fetched, the grid will automatically move to the 1st page.
+
+{{"demo": "RestoreStateInitialState.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Restore the state with `apiRef` [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+You can pass the state returned by `apiRef.current.exportState()` to the `apiRef.current.restoreState` method.
+In the demo below, clicking on **Save current view** will create a snapshot of the state that you can apply on the grid later through the **Custom view** menu.
```

Comment:
To avoid questions from the users. We only save what has changed. If I click on "Save current view" without making any change, then make a change, save again and restore the first saved snapshot it won't change anything. That's only how it works.

```suggestion
In the demo below, clicking on **Save current view** will create a snapshot of the changes made in the state, considering the initial state.
You can apply these changes on the grid later selecting a saved view in the **Custom view** menu.
```